src/sonic-host-services/scripts/caclmgrd

```diff
@@ -273,6 +274,38 @@ class ControlPlaneAclManager(daemon_base.DaemonBase):
 
         return allow_internal_docker_ip_cmds
 
+    def get_loopback_ip_from_loopback_keys(self, loopback_keys):
+
+        for key in loopback_keys:
+            if not _ip_prefix_in_key(key):
+                continue
+
+            iface_name, iface_cidr = key
+            if iface_name.startswith("Loopback3") and "/" in iface_cidr and "::" not in iface_cidr:
+                loopback_prefix = ipaddress.ip_network(str(iface_cidr))
+                loopback_ipv4 = loopback_prefix.split('/')[0]
+                loopback_address = str(loopback_ipv4)
+                return loopback_address
+
+        return None
+
+    def generate_fwd_traffic_from_host_to_soc(self, namespace):
+
+        fwd_dualtor_grpc_traffic_from_host_to_soc_cmds = []
+        if self.DualToR:
+            loopback_table = self.config_db_map[DEFAULT_NAMESPACE].get_table('LOOPBACK_INTERFACE')
+            loopback_address = self.get_loopback_ip_from_loopback_keys(loopback_table.keys())
+            if loopback_address is not None:
+                mux_table = self.config_db_map[DEFAULT_NAMESPACE].get_table(self.CONFIG_MUX_CABLE)
+                mux_table_keys = mux_table.keys()
+                for key in mux_table_keys:
+                    kvp = mux_table.get(key)
+                    if 'cable_type' in kvp and kvp['cable_type'] == 'active-active':
+                        fwd_dualtor_grpc_traffic_from_host_to_soc_cmds.append(self.iptables_cmd_ns_prefix[namespace] + 
+                                "iptables -t nat -A POSTROUTING --destination {} -j SNAT --to-source {}".format(kvp['soc_ipv4'], loopback_address))
```

Comment:
Can you please check with restarting caclmgrd? Does it fail or overwrite the existing rule?

Reply:
it seems it just adds the rule again,
target     prot opt source               destination         
SNAT       all  --  anywhere             192.168.0.7          to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.19         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.3          to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.21         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.23         to:10.1.0.38
SNAT       all  --  anywhere             co1speutlv104.phx.gbl  to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.5          to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.7          to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.9          to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.13         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.15         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.17         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.19         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.3          to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.21         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.23         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.25         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.7          to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.9          to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.11         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.13         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.15         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.17         to:10.1.0.38
SNAT       all  --  anywhere             win-cfr7mesbc6v.cheetah.vms  to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.3          to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.21         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.23         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.7          to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.11         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.13         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.15         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.21         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.23         to:10.1.0.38
SNAT       all  --  anywhere             192.168.0.25         to:10.1.0.38
admin@svcstr-7050-acs-1:~$